contexts/bounty-effects.tsx

```diff
@@ -5,23 +5,30 @@ import {useRouter} from "next/router";
 import {useAppState} from "contexts/app-state";
 
 import {useBounty} from "x-hooks/use-bounty";
+import useChain from "x-hooks/use-chain";
 
 const _context = {};
 
 export const BountyEffectsContext = createContext(_context);
 
 export const BountyEffectsProvider = ({children}) => {
-
-  const {query} = useRouter();
-  const {state} = useAppState();
   const bounty = useBounty();
+  const { chain } = useChain();
+  const { query } = useRouter();
+  const { state: { Service, connectedChain, currentBounty, currentUser } } = useAppState();
+
+  useEffect(bounty.getDatabaseBounty, [
+    chain,
+    query?.id,
+    query?.repoId,
+  ]);
 
-  useEffect(bounty.getDatabaseBounty, [state.Service?.network?.active, query?.id, query?.repoId]);
   useEffect(bounty.getChainBounty, [
-    state.Service?.active?.network?.contractAddress, 
-    state.Service?.network?.active?.networkAddress, 
-    state.currentBounty?.data?.contractId,
-    state.currentUser?.walletAddress 
+    Service?.active?.network?.contractAddress, 
+    Service?.network?.active?.networkAddress, 
+    currentBounty?.data?.contractId,
+    currentUser?.walletAddress,
+    connectedChain?.id,
```

Comment:
this is too much dependencies and it will query the chain _too much_ -- we need to either have a new dispatched value (isNetworkReady()) that is dispatched with `true` when `contractAddress` and `networkAddress` (and whatever more we usually need) are loaded in